orchagent/neighorch.cpp

```diff
@@ -1135,20 +1145,131 @@ void NeighOrch::doVoqSystemNeighTask(Consumer &consumer)
 
 bool NeighOrch::addInbandNeighbor(string alias, IpAddress ip_address)
 {
-    //For "port" type inband, the inband reachability info syncing can be done through static
-    //configureation or CHASSIS_APP_DB sync (this function)
+    //Add neighbor record in SAI without adding host route for local inband to avoid route
+    //looping for packets destined to the Inband interface if the Inband is port type
+
+    if(gIntfsOrch->isRemoteSystemPortIntf(alias))
+    {
+        //Remote Inband interface. Skip
+        return true;
+    }
+
+    sai_status_t status;
+    MacAddress inband_mac = gMacAddress;
+
+    sai_object_id_t rif_id = gIntfsOrch->getRouterIntfsId(alias);
+    if (rif_id == SAI_NULL_OBJECT_ID)
+    {
+        SWSS_LOG_INFO("Failed to get rif_id for %s", alias.c_str());
+        return false;
+    }
+
+    //Make the object key
+    sai_neighbor_entry_t neighbor_entry;
+    neighbor_entry.rif_id = rif_id;
+    neighbor_entry.switch_id = gSwitchId;
+    copy(neighbor_entry.ip_address, ip_address);
+
+    vector<sai_attribute_t> neighbor_attrs;
+    sai_attribute_t attr;
+    attr.id = SAI_NEIGHBOR_ENTRY_ATTR_DST_MAC_ADDRESS;
+    memcpy(attr.value.mac, inband_mac.getMac(), 6);
+    neighbor_attrs.push_back(attr);
+
+    if(!addVoqEncapIndex(alias, ip_address, neighbor_attrs))
```

Comment:
why we need this ?

Reply:
Usually for local neighbors we let the SAI allocate encap index and the neighbor along with the allocated encap index is synced to all the remote asics. Here we call the encap index addition for local neighbor to give the flexibility for user given encap index (user has to make sure that the encap index unique within asic). Simply configure the voq neighbor table with encap index attribute (similar to remote neighbor encap index) and that encap index will be sent to the SAI. So far we have not used this approach. Currently since we do not have encap index for local neighbors in voq neigh table, we always let the SAI allocate the encap index.
This api is similar to the addNeighbor() api that handles adding encap index for both local and remote neighbors except that this does not add next hop and the host route addition is masked (specifically required for inband port type neighbors)

Reply:
I don't understand 
if gIntfsOrch->isRemoteSystemPortIntf(alias) is True we return from this API. 
addVoqEncapIndex works only gIntfsOrch->isRemoteSystemPortIntf(alias) is True.
So why we are even calling it ?

Reply:
Good point. The addVoqEncapIndex() is supposed to get encap index from voq neighbor table only for remote system port neighbors. The addVoqEncapIndex() is called for all neihbors (local and remote neighbors) from addNeighbor() function. We need this remote system port interface check so that only for remote neighbors, encap index attribute will be sent to SAI. 
For addInbandNeighbor() since we are sure that this is a local neighbor, we can avoid calling addVoqEncapIndex() function. I'll fix this.

Reply:
Has this been fixed ?

Reply:
Yes. Fixed.